pkg/reconciler/buildrun/resources/sources/utils.go

```diff
@@ -105,6 +105,16 @@ func AppendWriteableVolumes(
 	)
 	// Point the TMPDIR environment variable to the custom path.
 	setEnvVar(targetStep, "TMPDIR", tmpDir)
+
+	// Define a custom, isolated path for the home directory and mount it.
+	writeableHomeMountPath := "/shp-writable-home"
+	addStepEmptyDirVolume(
+		taskSpec,
+		targetStep,
+		generateVolumeName("shp-home-", targetStep.Name),
+		writeableHomeMountPath,
+	)
+	setEnvVar(targetStep, "HOME", writeableHomeMountPath)
```

Comment:
/hold

I think this might break/interfere with Tekton's home directory feature. There is a flag in the operator to disable this behavior. https://tekton.dev/docs/operator/tektonpipeline/

Unfortunately this feature of Tekton does not appear to be well documented.

Reply:
/hold cancel

Turns out this was an old feature in Tekton, removed a long time ago: https://github.com/tektoncd/pipeline/pull/4587